Google.Api.Generator/Generation/EnumStringGenerator.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Generator.ProtoUtils;
+using Google.Api.Generator.Utils;
+using Google.Api.Generator.Utils.Roslyn;
+using Google.Protobuf.Reflection;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using static Google.Api.Generator.Utils.Roslyn.Modifier;
+using static Google.Api.Generator.Utils.Roslyn.RoslynBuilder;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Google.Api.Generator.Generation;
+
+/// <summary>
+/// Generates (to stdout) a static class (with nested static classes) to provide
+/// string constants for the wire representations of enum values. This is only used
+/// for DIREGAPIC APIs.
+/// </summary>
+internal class EnumStringGenerator
+{
+    // TODO: Determine a policy or config for the class, file and API description.
+    internal const string RootClassName = "ComputeEnumConstants";
+    private const string ApiDescription = "Compute API";
+
+    public static CompilationUnitSyntax MaybeGenerate(SourceFileContext ctx, string ns, IEnumerable<FileDescriptor> packageFileDescriptors)
+    {
+        // TODO: Add an option in the service config for this... although 
```

Comment:
nit: seems incomplete... although I'll probably be happy with what comes after the although :)